server/py/framework/utils/clients/messaging.py

```diff
@@ -40,8 +41,106 @@ def __init__(self) -> None:
         self._session: typing.Optional[mlrun.utils.AsyncClientWithRetry] = None
         # Retry session is for internal messaging
         self._retry_session: typing.Optional[mlrun.utils.AsyncClientWithRetry] = None
```

Comment:
not really related, but it seems we don't use `_retry_session` anywhere.

Also, general question related to the bug from this pr https://github.com/mlrun/mlrun/pull/7324. Should `_sync_retry_session` be per thread as well?

Reply:
wrt `_retry_session` @moranbental commented about it as well above and I explained why I added it there.
Regarding `_sync_retry_session` I believe so. I think if your fix works we should change the sync session as well

Reply:
add comment that we don't use it and what is the future purpose of it